### ISSUE TITLE
command line feature request

### ISSUE THREAD

**Issue by nsrderooy** [hardinfo2/hardinfo2#46]
Hi, first of all thanks for this tool.

I would like to make a (few) feature request(s) for the command line options

I think it would be helpful if these features are added

`--list-benchmark` lists all benchmarks useful for `--run-benchmark` or `--skip-benchmark`
`--run-benchmark` 'all' to run all benchmarks instead of just one
`--synchronize` Synchronize before starting
`--upload` synchronize after generating report or running benchmark
`--accept` accept uploading report to server without confirmation

also for the GUI it would be nice to have a button to run all benchmarks at once.

Thanks in advance.

**Comment by hwspeedy** [comment]
Thanx for feedback, it's much appreciated.

You have some good suggestions and also show that our explanations could be better.

**Where to find help for params:**
hardinfo2 --help
man hardinfo2
https://hardinfo2.org/userguide - FAQ and params

**Missing CLI features:**
`-l --list-benchmarks`  lists all benchmarks

**Help to the other situations:**
To run all benchmarks in CLI:
avoid adding the -b switch - default is all benchmarks are run.

To Synchronize in CLI before starting, run all benchmarks and submit your benchmark without any further actions needed:
hardinfo2 -r -u 1

The skip one benchmark might not be needed - it can be done by adding all wanted instead (positive list)

To run all benchmarks in GUI, press the synchronize button.

**Testing new feature - Group Benchmark Feature**
To benchmark a group of your machines from GUI or CLI and get only those shown on https://hardinfo2.org.

**CLI**
run on all machines: hardinfo2 -r -u MyUniqueUserNote 
go to web page: https://hardinfo2.org/?u=MyUniqueNote

**GUI**
start program like hardinfo2 -u MyUniqueUserNote and press the synchronize button in GUI.
go to web page: https://hardinfo2.org/?u=MyUniqueUserNote

UserNote: may consists of letters and numbers

Please provide feedback and ideas to this new group benchmarking feature. Problem I see is that your benchmarks could easily be "polluted" by others...
But that is also a nice feature - that a group of persons all use the same usernote tag and compares against another. Still they might want to could erase one unwanted from group.... - Ideas are welcome..

**Comment by nsrderooy** [comment]
Thanks for the feedback I checked both the man page and `--help` but initially focust on the `--help` option.
But now I see that there are discrepancies between the man page and the `--help`  option;

man page
>        -b, --run-benchmark
>              run a specific benchmark eg. -b 'FPU FFT'  (Default all benchmarks runs when generate report)

`--help`
>   -b, --run-benchmark       run single benchmark eg. -b 'FPU FFT' (Default all Benchmarks runs)

The `--help` is actually not true or I might not understand it correctly, as it shows this error;

`hardinfo2 --run-benchmark`
> Unrecognized arguments.
> Try ``hardinfo2 --help'' for more information.

also the man page is missing the `-w, --very-verbose        show all details`

**Comment by nsrderooy** [comment]
Sorry, wrong button...

**Comment by hwspeedy** [comment]
Some alignment between the 3 places to get params help:

man page - add -w:
https://github.com/hardinfo2/hardinfo2/commit/a2b943993c46a8d84dccd85d3bbb7343ab85b8da

userguide - add -w:
https://github.com/hardinfo2/server/commit/a5e84cfc765687f34593cbd6e8c97fe7e095fa1d

**Comment by yankee495** [comment]
Hello,

Since you're working on the command line here, I thought I'd report a small bug.
When I run "Sysbench CPU Multi-thread" it says it's running "sysbench memory benchmark (Multi-thread)". But it is in fact running Sysbench CPU Multi-thread.
I just found this tonight.

I'm running Hardinfo2 2.2.7 for Fedora on openSUSE Tumbleeed for testing.
Tumbleweed's 2.2.4 had the problem so I tried the newest version to see if it existed in 2.2.7.
It does.

```
> hardinfo2 -b 'SysBench CPU (Multi-thread)'
 Performing Alexey Kopytov's sysbench memory benchmark (Multi-thread)...
95178.531250; 7.000300; 32; 1000019; --time=7 --cpu-max-prime=10000
```
I found this while checking a regression between Sysbench 1.0.19 and Sysbench 1.0.20.
In Sysbench 1.0.20 my Sysbench CPU Multi-thread score dropped from 95,000 to 87,000.
I installed Sysbench 1.0.19 from Leap 15.2 and my score went back to 95,000.
You may get tons of bug reports but it is in Sysbench, or just my system.
I ran Sysbench directly in the terminal so this drop has nothing to do with Hardinfo2.

**Comment by hwspeedy** [comment]
Just added a CLI command to search via topics to show only part of info:

https://github.com/hardinfo2/hardinfo2/commit/3b5318a6b488260cdab539e24cb6f55e7bc430f3

PS: Only available in master builds

**Usage**
1. Show topics: Main, submain, parts
hardinfo2 -t getlist 

2. examples
hardinfo2 -t proc     (Short for processor)
![Image](https://github.com/user-attachments/assets/e258de77-440e-421a-b24b-a850edcd650b)

hardinfo2 -t stor     (Short for storage)
hardinfo2 -t pci
hardinfo2 -t usb
hardinfo2 -t net       (Short for network)

PS: Case insensitive

**Comment by hwspeedy** [comment]
See #152 for further discussion.

Closing this one as done for now.

PS: User guide will be updated with new features, when hardinfo2 is released next time.
